From 9c470a0460a0350cdad7beb13568948b29e01f76 Mon Sep 17 00:00:00 2001
From: ruscher <rruscher@gmail.com>
Date: Mon, 26 Oct 2026 18:59:22 -0300
Subject: [PATCH 25/30] wlclient: Fixes inserting a timer before other
 registered timers.

wlmcl_client_timer_create() did not stop iterating after inserting the
new timer before a timer that is due later: It inserted the same node
again before each further such timer, and finally appended it at the
end of the list. As soon as a timer was registered while another one,
due later, was pending, the timer list got corrupted.

Stops iterating once the timer is inserted.

Signed-off-by: ruscher <rruscher@gmail.com>
---
 src/wlclient/wlclient.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/wlclient/wlclient.c b/src/wlclient/wlclient.c
index 9f38c95..e64b4c9 100644
--- a/src/wlclient/wlclient.c
+++ b/src/wlclient/wlclient.c
@@ -889,6 +889,7 @@ wlmcl_client_timer_t *wlmcl_client_timer_create(
         if (timer_ptr->target_usec > ref_timer_ptr->target_usec) continue;
         bs_dllist_insert_node_before(
             &client_ptr->timers, dlnode_ptr, &timer_ptr->dlnode);
+        break;
     }
     if (NULL == dlnode_ptr) {
         bs_dllist_push_back(&client_ptr->timers, &timer_ptr->dlnode);
-- 
2.55.0

